src/Faker/Provider/Base.php

```diff
@@ -192,6 +192,10 @@ public static function randomElements($array = ['a', 'b', 'c'], $count = 1, $all
         $traversables = [];
         $arr = $array;
 
+        if (is_string($array) && function_exists('enum_exists') && enum_exists($array)) {
+            $traversables = $array::cases();
```

Comment:
If we change this to `$arr = $traversables = $array::cases()` the test passes, but I'm still unsure why it was failing in the first place